File: ObjectiveCiOSProject/MoMagicNotificationServices/NotificationService.m
//
//  NotificationService.m
//  MoMagicNotificationServices
//
//  Created by Amit on 14/06/21.
//

#import "NotificationService.h"

@interface NotificationService ()

@property (nonatomic, strong) void (^contentHandler)(UNNotificationContent *contentToDeliver);
@property (nonatomic, strong) UNMutableNotificationContent *bestAttemptContent;
@property (nonatomic, strong) UNNotificationRequest *receivedRequest;

@end

@implementation NotificationService

- (void)didReceiveNotificationRequest:(UNNotificationRequest *)request withContentHandler:(void (^)(UNNotificationContent * _Nonnull))contentHandler {
    self.contentHandler = contentHandler;
    self.receivedRequest = request;
    self.bestAttemptContent = [request.content mutableCopy];
    if (self.bestAttemptContent != nil)
    {
        [DATB didReceiveNotificationExtensionRequestWithBundleName:@"com.momagic.ObjectiveCiOSProject" soundName:@"" isBadge:true request:self.receivedRequest bestAttemptContent:self.bestAttemptContent contentHandler:self.contentHandler];
    }
}
- (void)serviceExtensionTimeWillExpire {
   
    [DATB didReceiveNotificationExtensionRequestWithBundleName:@"com.momagic.ObjectiveCiOSProject" soundName:@"" isBadge:true request:self.receivedRequest bestAttemptContent:self.bestAttemptContent contentHandler:self.contentHandler];
}

@end

File: MoMagiciOSProject/MoMagicExtesnionServices/NotificationService.swift
//
//  NotificationService.swift
//  MoMagicExtesnionServices
//
//  Created by Amit on 15/06/21.
//

import UserNotifications
import MomagiciOSSDK
import AVFoundation

class NotificationService: UNNotificationServiceExtension {
    var bombSoundEffect: AVAudioPlayer?

    var contentHandler: ((UNNotificationContent) -> Void)?
      var bestAttemptContent: UNMutableNotificationContent?
      var receivedRequest: UNNotificationRequest!
      override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
          self.receivedRequest = request;
          self.contentHandler = contentHandler
          bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent)
          if let bestAttemptContent = bestAttemptContent {
//              DATB.didReceiveNotificationExtensionRequest(bundleName:"" , soundName: "com.momagic.MoMagiciOSProject", request: "", bestAttemptContent: bestAttemptContent, contentHandler: contentHandler)
              
              DATB.didReceiveNotificationExtensionRequest(bundleName: "com.momagic.MoMagiciOSProject", soundName: "String", request: receivedRequest, bestAttemptContent: bestAttemptContent, contentHandler: contentHandler)
         
          
           
        }
        }
        override func serviceExtensionTimeWillExpire() {
          if let contentHandler = contentHandler, let bestAttemptContent = bestAttemptContent {
              DATB.didReceiveNotificationExtensionRequest(bundleName: "String", soundName: "String", request: receivedRequest, bestAttemptContent: bestAttemptContent, contentHandler: contentHandler)
              
          }
        }

}
